pkg/crd/schema.go

```diff
@@ -404,7 +404,7 @@ func structToSchema(ctx *schemaContext, structType *ast.StructType) *apiext.JSON
 		Properties: make(map[string]apiext.JSONSchemaProps),
 	}
 
-	if ctx.info.RawSpec.Type != structType {
+	if ctx.info.RawSpec != nil && ctx.info.RawSpec.Type != structType {
```

Comment:
But this would allow embedded structs / anonymous structs, which does not seem to be allowed according to the error below?

Shouldn't this then be:

```suggestion
	if ctx.info.RawSpec == nil || ctx.info.RawSpec.Type != structType {
```